### PR TITLE
fix(gradle): version compatibility

### LINKED ISSUE #5424: Dependency Extraction Fails for Gradle < 6
As reported in https://github.com/renovatebot/renovate/issues/4471:

The new gradle extraction script depends on a type that was introduced with Gradle 6, leading to error like this when it is executed on against a project with a Gradl 5 wrapper:

```
FAILURE: Build failed with an exception.\n\n* Where:\nInitialization script '/mnt/renovate/gh/moltenbits/renovate-gradle-bug/renovate-plugin.gradle' line: 4\n\n* What went wrong:\nCould not compile initialization script '/mnt/renovate/gh/moltenbits/renovate-gradle-bug/renovate-plugin.gradle'.\n> startup failed:\n  initialization script '/mnt/renovate/gh/moltenbits/renovate-gradle-bug/renovate-plugin.gradle': 4: unable to resolve class org.gradle.api.artifacts.repositories.UrlArtifactRepository\n   @ line 4, column 1.\n     import org.gradle.api.artifacts.repositories.UrlArtifactRepository
```

### PR DESCRIPTION
**Pull request by jGleitz**
Fixes the use of a type that was introduces in Gradle 6.

Adds basic Gradle compatibility tests, more testing is needed! (#5408)

Closes #5424

### CODE REVIEW

**Comment by jGleitz** [comment]
@JamieMagee @ViceIce we should settle for one implementation. Check for `URI` or for `MavenArtifactRepository`?

**Comment by jGleitz** [comment]
Note that the code declares all dependencies to be maven dependencies, so checking for `MavenArtifactRepository` does not exclude any dependencies.

**Comment by ViceIce** [comment]
I think we can go with `MavenArtifactRepository`. I don't know if there is any other repository type in the wild.

If there is, we can add it explicitly later.

**Review by viceice — DISMISSED** [review]

**Comment by jGleitz** [comment]
> I don't know if there is any other repository type in the wild.

There is: [`IvyArtifactRepository`](https://docs.gradle.org/current/javadoc/org/gradle/api/artifacts/repositories/IvyArtifactRepository.html). But since renovate can’t read from it, it doesn’t matter that we exclude it.

**Comment by ViceIce** [comment]
Then we should open a new issue for ivy, just to remember it.

**Comment by jGleitz** [comment]
I am a little stuck with this PR because I don’t know how to set up Java for Circle CI. Should I build and use a new Docker image?

Also, why is CircleCI still used in the first place? Most checking is done on GitHub, but one job is left on CircleCI.

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -33,6 +33,11 @@ jobs:
         with:
           python-version: ${{ matrix.python-version }}
 
+      - name: Set up Java 8
+        uses: actions/setup-java@v1
+        with:
+          java-version: 8
```
Comment (ViceIce) [inline .github/workflows/test.yml]: We should use latest java lts (11), like our renovate:gradle images.

**Review by viceice — CHANGES_REQUESTED** [review]

**Comment by ViceIce** [comment]
We can skip circle java tests for now, as we plan to move to github actions soon

**Comment by jGleitz** [comment]
Alright. I think this PR should be ready then.

**Review by viceice — DISMISSED** [review]

**Comment by ViceIce** [comment]
maybe we should also skip the java tests on windows.

**Comment by rarkins** [comment]
Does this mean that Java is needed locally to pass `yarn test`? If so, we should probably update the docs about it.

File: lib/manager/gradle/gradle-updates-report.spec.ts [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+const fixtures = 'lib/manager/gradle/__fixtures__';
+const skipJavaTestsEnv = 'SKIP_JAVA_TESTS';
+
+const gradleJavaVersionSupport = {
+  5: { min: 8, max: 12 },
+  6: { min: 8, max: 13 },
+};
+
+function parseJavaVersion(javaVersionOutput) {
```
Comment (JamieMagee) [inline lib/manager/gradle/gradle-updates-report.spec.ts]:
```suggestion
function parseJavaVersion(javaVersionOutput: string) {
```

**Review by JamieMagee — APPROVED** [review]

**Pull request merged**